pyomo/solvers/tests/checks/test_cplex.py

```diff
@@ -54,5 +59,198 @@ def test_validate_file_name(self):
+
+    def test_write_priority_to_priorities_file(self):
+        self.mock_model.priority = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        priority_val = 10
+        self.mock_model.priority.set_value(self.mock_model.x, priority_val)
+
+        CPLEXSHELL._write_priorities_file(self.mock_cplex_shell, self.mock_model)
+        priorities_file = self.get_priorities_file_as_string(self.mock_cplex_shell)
+
+        self.assertEqual(
+            priorities_file,
+            "* ENCODING=ISO-8859-1\nNAME             Priority Order\n  x1 %d\nENDATA\n"
+            % (priority_val,),
+        )
+
+    def test_write_priority_and_direction_to_priorities_file(self):
+        self.mock_model.priority = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        priority_val = 10
+        self.mock_model.priority.set_value(self.mock_model.x, priority_val)
+
+        self.mock_model.direction = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        direction_val = BranchDirection.down
+        self.mock_model.direction.set_value(self.mock_model.x, direction_val)
+
+        CPLEXSHELL._write_priorities_file(self.mock_cplex_shell, self.mock_model)
+        priorities_file = self.get_priorities_file_as_string(self.mock_cplex_shell)
+
+        self.assertEqual(
+            priorities_file,
+            "* ENCODING=ISO-8859-1\nNAME             Priority Order\n %s x1 %d\nENDATA\n"
+            % (ORDFileSchema._direction_to_str(direction_val), priority_val),
+        )
+
+    def test_raise_due_to_invalid_priority(self):
+        self.mock_model.priority = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        self.mock_model.priority.set_value(self.mock_model.x, -1)
+        with self.assertRaises(ValueError):
+            CPLEXSHELL._write_priorities_file(self.mock_cplex_shell, self.mock_model)
+
+        self.mock_model.priority.set_value(self.mock_model.x, 1.1)
+        with self.assertRaises(ValueError):
+            CPLEXSHELL._write_priorities_file(self.mock_cplex_shell, self.mock_model)
+
+    def test_use_default_due_to_invalid_direction(self):
+        self.mock_model.priority = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        priority_val = 10
+        self.mock_model.priority.set_value(self.mock_model.x, priority_val)
+
+        self.mock_model.direction = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        self.mock_model.direction.set_value(
+            self.mock_model.x, "invalid_branching_direction"
+        )
+
+        CPLEXSHELL._write_priorities_file(self.mock_cplex_shell, self.mock_model)
+        priorities_file = self.get_priorities_file_as_string(self.mock_cplex_shell)
+
+        self.assertEqual(
+            priorities_file,
+            "* ENCODING=ISO-8859-1\nNAME             Priority Order\n  x1 %d\nENDATA\n"
+            % (priority_val,),
+        )
+
+
+class CPLEXShellSolvePrioritiesFile(unittest.TestCase):
```

Comment:
Why is this a separate class from the previous priorities tester?

Reply:
The former class is a unit test specifically on the writing of priorities via `_write_priorities_file()`. The latter is an integration test on the end-to-end application of priorities via the `Suffix` through a `solve()`. I've added docs to that effect